src/sql/workbench/parts/query/browser/queryEditor.ts

```diff
@@ -211,66 +291,68 @@ export class QueryEditor extends BaseEditor {
 		}
 	}
 
+
 	/**
 	 * Called to indicate to the editor that the input should be cleared and resources associated with the
 	 * input should be freed.
 	 */
 	public clearInput(): void {
-		if (this._resultsEditor) {
-			this._resultsEditor.clearInput();
-		}
-		if (this._sqlEditor) {
-			this._sqlEditor.clearInput();
-		}
-		this._disposeEditors();
+		this.textEditor.clearInput();
+		this.resultsEditor.clearInput();
 		super.clearInput();
 	}
 
 	/**
 	 * Sets focus on this editor. Specifically, it sets the focus on the hosted text editor.
 	 */
 	public focus(): void {
-		if (this._sqlEditor) {
-			this._sqlEditor.focus();
-		}
+		this.textEditor.focus();
 	}
 
 	/**
 	 * Updates the internal variable keeping track of the editor's size, and re-calculates the sash position.
 	 * To be called when the container of this editor changes size.
 	 */
 	public layout(dimension: DOM.Dimension): void {
-		this._dimension = dimension;
-
-		if (this._sash) {
-			this._setSashDimension();
-			this.sash.layout();
-		}
-
-		this._doLayout();
-		this._resizeGridContents();
+		this.dimension = dimension;
+		this.splitview.layout(dimension.height - 31);
 	}
 
 	/**
 	 * Returns the editor control for the text editor.
 	 */
 	public getControl(): IEditorControl {
-		if (this._sqlEditor) {
-			return this._sqlEditor.getControl();
-		}
-		return null;
+		return this.textEditor.getControl();
 	}
 
-	public getQueryResultsEditor(): QueryResultsEditor {
-		return this._resultsEditor;
+	public setOptions(options: EditorOptions): void {
+		this.textEditor.setOptions(options);
 	}
 
-	public getSqlEditor(): TextResourceEditor {
-		return this._sqlEditor;
+	private removeResultsEditor() {
+		if (this.resultsVisible) {
+			this.splitview.removeView(1, Sizing.Distribute);
+			this.resultsVisible = false;
+		}
+	}
+
+	private addResultsEditor() {
+		if (!this.resultsVisible) {
+			this.splitview.addView({
+				element: this.resultsEditorContainer,
+				layout: size => this.resultsEditor && this.resultsEditor.layout(new DOM.Dimension(this.dimension.width, size)),
+				minimumSize: 220,
```

Comment:
>220 [](start = 17, length = 3)

How did you pick 220? Is it tied to the min size of the text editor container above? If so would suggest using a single value for them

Reply:
Its basically arbitrary, I just found the number in other places using this strategy. I figured we would play with it as people were using the new code.